tests/zfs-tests/tests/functional/delegate/cleanup.ksh

```diff
@@ -42,6 +43,8 @@ if ! is_linux; then
 	fi
 fi
 
-default_cleanup
+if is_linux; then
```

Comment:
The use of `is_linux` here and throughout isn't needed, since the library function makes an OS check as well.

Reply:
@jwk404 I thought that SunOS doesn't have `zfs_admin_snapshot` parameter, and this code will fail on it https://github.com/zfsonlinux/zfs/blob/master/tests/zfs-tests/include/libtest.shlib#L3469 , no?

I'd prefer to have `is_linux` check explicitly.

https://github.com/illumos/illumos-gate/search?utf8=%E2%9C%93&q=admin_snapshot&type=